tests/bluetooth/host/keys/bt_keys_foreach_bond/src/main.c

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/addr.h>
+#include <host/keys.h>
+#include "mocks/keys_help_utils.h"
+#include "testing_common_defs.h"
+
+/* This LUT contains different combinations of ID and Address pairs */
+static const struct id_addr_pair testing_id_addr_pair_lut[] = {
+	{ BT_ADDR_ID_1, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_1, BT_ADDR_LE_2 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_1 },
+	{ BT_ADDR_ID_2, BT_ADDR_LE_2 }
+};
+
+/* This list will hold returned references while filling keys pool */
+static struct bt_keys *returned_keys_refs[CONFIG_BT_MAX_PAIRED];
+
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == CONFIG_BT_MAX_PAIRED);
+BUILD_ASSERT(ARRAY_SIZE(testing_id_addr_pair_lut) == ARRAY_SIZE(returned_keys_refs));
+
+/* Callback to be used when no calls are expected by bt_foreach_bond() */
+static void bt_foreach_bond_unreachable_cb(const struct bt_bond_info *info, void *user_data)
+{
+	zassert_unreachable("Unexpected call to '%s()' occurred", __func__);
+}
+
+/* Callback to be used when calls are expected by bt_foreach_bond() */
+static void bt_foreach_bond_expected_cb(const struct bt_bond_info *info, void *user_data)
+{
+	uint32_t *call_counter = (uint32_t *)user_data;
+
+	zassert_true(info != NULL, "Unexpected NULL reference pointer for parameter '%s'", "info");
+	zassert_true(user_data != NULL, "Unexpected NULL reference pointer for parameter '%s'",
+		     "user_data");
+
+	(*call_counter)++;
+}
+
+static void *type_not_set_ts_setup(void)
+{
+	clear_key_pool();
+	int rv = fill_key_pool_by_id_addr(testing_id_addr_pair_lut,
+					  ARRAY_SIZE(testing_id_addr_pair_lut), returned_keys_refs);
+
+	zassert_true(rv == 0, "Failed to fill keys pool list, error code %d", -rv);
+
+	return NULL;
+}
+
+ZTEST_SUITE(bt_keys_foreach_bond_keys_type_not_set, NULL, type_not_set_ts_setup, NULL, NULL, NULL);
```

Comment:
the two test suites can be merged IMO (e.g. name `bt_keys_foreach_bond`). You can move the setup from test suite setup to test case setup.

Reply:
no, the test case function should be executed once before each test case, that's the whole point. It's the test suite function that is executed once per suite.